src/Perlang.Interpreter/Compiler/PerlangCompiler.cs

```diff
@@ -573,6 +573,11 @@ private void RegisterGlobalClasses()
 
                     // Probably enabled by -Wconversion, but this causes issues with valid code like `2147483647 /
                     // 4294967295U`.
+                    //
+                    // NOTE: disabling this also enables extremely dangerous constructs like being able to call
+                    // BigInt::pow(-3), causing the negative integer to be reinterpreted as a very large uint32 instead.
+                    // We should re-think this and really try to re-enable this, to prevent such horrible code from
+                    // compiling.
```

Comment:
:point_up: The above was when `exponent` was a `uint32_t`. I tried to capture here some of the dangers of our current semantics. Not intending to do anything about this at the moment, just writing it down for future remembrance.